subsys/bluetooth/mesh/shell/shell_scene_cli.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+
+static void scene_get_print(const struct shell *shell, int err, struct bt_mesh_scene_state rsp)
+{
+	if (!err) {
+		shell_print(shell,
+			    "Status: %d, current scene: %d "
+			    "target scene: %d, rem time: %d",
+			    rsp.status, rsp.current, rsp.target, rsp.remaining_time);
+	}
+}
+
+static int cmd_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SCENE_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_scene_cli *cli = mod->user_data;
+	struct bt_mesh_scene_state rsp;
+
+	int err = bt_mesh_scene_cli_get(cli, NULL, &rsp);
+
+	scene_get_print(shell, err, rsp);
+	return err;
+}
+
+static void scene_print(const struct shell *shell, int err, struct bt_mesh_scene_register rsp)
+{
+	if (!err) {
+		shell_print(shell, "Status: %d, current scene: %d", rsp.status, rsp.current);
+
+		for (int i = 0; i < rsp.count; i++) {
+			shell_print(shell, "Scene %d: %d", i, rsp.scenes[i]);
+		}
+	}
+}
+
+static int cmd_register_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint16_t cnt = (uint16_t)strtol(argv[1], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SCENE_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_scene_cli *cli = mod->user_data;
+	uint16_t scenes[CONFIG_BT_MESH_SCENES_MAX];
+	struct bt_mesh_scene_register rsp = {
+		.count = CONFIG_BT_MESH_SCENES_MAX,
+		.scenes = &scenes[0],
+	};
+
+	int err = bt_mesh_scene_cli_register_get(cli, NULL, &rsp);
+
+	scene_print(shell, err, rsp);
+	return err;
+}
+
+static int store(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	uint16_t scene = (uint16_t)strtol(argv[1], NULL, 0);
```

Comment:
Well, you are right. I guess strtol could be wraped in utilities with cheks that ensure that the input is numeric. Would you be comfortable If i create a separate task for this? It would have to be applied to the rest of the shell cli models aswell.